### PR TITLE
feat: verify first, check semantics after and remove issuanceDate check

### PR DESCRIPTION
**Pull request by lleifermann**
This PR proposes the some changes as discussed in: https://github.com/digitalcredentials/vc/issues/7

- Verify Presenation Proof & Credential Proof **_FIRST_** **before** doing any semantic checks as the fields may be tampered with. 
- Remove check if issuanceDate is not in the future as this is a fully expected use-case (to issue credentials that become valid at some point in time)

IMHO we should also remove the check (if not even all of them) if a credential is expired when issuing it. For some e2e test scenarios one may choose to issue an expired credential.

I am still a bit confused. Initally my PR was larger, with more modifications to error handling as well. But it seems jsigs already does alot of sanity and semantic checks on the credential (checking if a context is there for example). Is there a reason we do them again here?

### CODE REVIEW

**Comment by lleifermann** [comment]
Actually - I think it makes sense to separate the checks for issuance and verification. As for verification a check on the issuanceDate may be useful.

**Comment by lleifermann** [comment]
I also highly recommend adding github actions to this repo. I ran all the tests locally but that seems like such a thing of the past 😄

File: lib/index.js [hunk 1]
```diff
@@ -329,6 +320,16 @@ async function _verifyCredential(options = {}) {
     return result;
   }
 
+  // run common credential checks (add check results to log)
+  _checkCredential({credential, log, now});
+
+  // if credential status is provided, a `checkStatus` function must be given
+  if(credential.credentialStatus && typeof options.checkStatus !== 'function') {
```
Comment (dmitrizagidulin) [inline lib/index.js:327]: Would it make sense to move this check to be fore signature verification?

Comment (lleifermann) [reply lib/index.js:327]: Idea was that the first motivation of a verification should always be checking the proof. If this is skipped for whatever reason (some method of field of the credential missing) we would exit without knowing if the credential has been tampered with. Not sure if this is the best approach.

**Comment by lleifermann** [comment]
@dmitrizagidulin Any update on this?

**Review by dmitrizagidulin — APPROVED** [review]
:+1: 
@lleifermann thanks again!

**Comment by dmitrizagidulin** [comment]
@lleifermann - published as `@digitalcredentials/vc` v5.0.0

**Pull request merged**